test/CMakeLists.txt

```diff
@@ -4,12 +4,23 @@ function(addtest name)
     COMMAND ${CMAKE_SOURCE_DIR}/test/run ${name}
     WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
   )
-  set(environment CCACHE=${CMAKE_BINARY_DIR}/ccache EXIT_IF_SKIPPED=true)
-  set_tests_properties(
-    "producttest.${name}"
-    PROPERTIES
-    ENVIRONMENT "${environment}"
-    SKIP_RETURN_CODE 125)
+  set_tests_properties("producttest.${name}" PROPERTIES
+    ENVIRONMENT "CCACHE=${CMAKE_BINARY_DIR}/ccache;EXIT_IF_SKIPPED=true")
```

Comment:
(I extracted the environment list to a separate variable since the string list variable syntax felt cleaner.)

Reply:
Ah sorry, I was confused by set environment and then set tests properties environment. Once I understood what was going on I thought I would make it simpler :-)